### PR TITLE
add simdCsvReader close in external scan

### PR DESCRIPTION
**Pull request by jianwan0214**
## What type of PR is this?

- [ ] API-change
- [x] BUG
- [ ] Improvement
- [ ] Documentation
- [ ] Feature
- [ ] Test and CI
- [ ] Code Refactoring

## Which issue(s) this PR fixes:

issue #5142

## What this PR does / why we need it:
add simdCsvReader close in externale scan

### CODE REVIEW

**Review by daviszhen — APPROVED** [review]

**Review by yingfeng — APPROVED** [review]

File: pkg/sql/colexec/external/external.go [hunk 1]
```diff
@@ -67,7 +67,6 @@ func Prepare(proc *process.Process, arg any) error {
 
 	if len(fileList) == 0 {
 		param.End = true
-		return moerr.NewInternalError("no such file '%s'", param.extern.Filepath)
```
Comment (xzxiong) [inline pkg/sql/colexec/external/external.go:70]: still needs warning log.

Comment (xzxiong) [reply pkg/sql/colexec/external/external.go:70]: seems already done. ~_~. lgtm

**Review by xzxiong — APPROVED** [review]

File: pkg/sql/colexec/merge/merge.go [hunk 2]
```diff
@@ -41,7 +42,12 @@ func Call(idx int, proc *process.Process, arg any) (bool, error) {
 			return true, nil
 		}
 		reg := proc.Reg.MergeReceivers[ap.ctr.i]
-		bat := <-reg.Ch
+		var bat *batch.Batch = nil
+		select {
+		case <-reg.Ctx.Done():
+		case bat = <-reg.Ch:
+		}
+
```
Comment (fengttt) [inline pkg/sql/colexec/merge/merge.go:50]: Why do you need this?   Did old code hang, without this?

**Review by fengttt — APPROVED** [review]
Did we see any improvement?

**Pull request merged**